common/display/displayqueue.cpp

```diff
@@ -149,27 +149,35 @@ bool DisplayQueue::ApplyPendingModeset(drmModeAtomicReqPtr property_set) {
   return true;
 }
 
-bool DisplayQueue::SetDpmsMode(uint32_t dpms_mode) {
+bool DisplayQueue::SetPowerMode(uint32_t power_mode) {
   ScopedSpinLock lock(spin_lock_);
-  if (dpms_mode_ == dpms_mode)
+
+  if (dpms_mode_ == power_mode)
     return true;
 
-  dpms_mode_ = dpms_mode;
-  if (dpms_mode_ == DRM_MODE_DPMS_OFF) {
+  if ((power_mode == kOff) || (power_mode == kDoze)) {
+    dpms_mode_ = DRM_MODE_DPMS_OFF;
+    Flush();
```

Comment:
The only worry I have is depending on the queue the power off/doze can be delayed a lot (If u think each update takes 16.6 msec and if we have two in queue than its 2* 16.6). If the action needs to be immediate than we need to do it immediately. But I think we need to test this out and see how it feels. Thats why a demo would be great to check out.